python/pyiceberg/table/__init__.py

```diff
@@ -152,6 +154,27 @@ def set_properties(self, **updates: str) -> Transaction:
         """
         return self._append_updates(SetPropertiesUpdate(updates=updates))
 
+    def add_column(
```

Comment:
I don't think we want to have this API on the Transaction itself. I like what @rdblue is suggesting [here](https://github.com/apache/iceberg/pull/6323#discussion_r1217020366).

This would look like:
```python
# When you want to do multiple things
with table.transaction() as t:
    t.updateSchema().addColumn("x", IntegerType()).commit()
    t.set_property(updated_at=str(datetime.now())

# or when you don't want to use a transaction:
t.updateSchema().addColumn("x", IntegerType()).commit()    
```

The `.commit()` will accumulate all the changes to the schema, and combine this into a single `AddSchemaUpdate` that's being added to the `self._updates`. This way we don't need `self._schema_update` which seems to be a little brittle since we set it to null again.

Reply:
I have made changes and currently support both of the above usage.

But do we need to support the following way?

```
txn = table.transaction()
txn.update_schema().add_column(name="x", type_var=IntegerType(), doc="doc")
txn.commit_transaction()
````

Reply:
Thinking of it, I think the transaction needs to be on a catalog level.